ethers-core/src/types/trace/geth/call.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{
+    types::{Address, Bytes, NameOrAddress, H256, U256},
+    utils::from_int_or_hex,
+};
+use serde::{Deserialize, Serialize};
+
+// https://github.com/ethereum/go-ethereum/blob/91cb6f863a965481e51d5d9c0e5ccd54796fd967/eth/tracers/native/call.go#L44
+#[derive(Debug, Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct CallFrame {
+    #[serde(rename = "type")]
+    pub typ: String,
+    pub from: Address,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub to: Option<NameOrAddress>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub value: Option<U256>,
+    #[serde(default, deserialize_with = "from_int_or_hex")]
+    pub gas: U256,
+    #[serde(default, deserialize_with = "from_int_or_hex", rename = "gasUsed")]
+    pub gas_used: U256,
```

Comment:
The `gas` and `gas_used` fields are not compatible with [the legacy format](https://github.com/ethereum/go-ethereum/blob/master/eth/tracers/internal/tracetest/testdata/call_tracer_legacy/simple.json#LL60C5-L60C5). Should we change it to `Option`, and which might be a break change?      
For now, to be compatible with previous versions, I've only added a default handling, which returns `U256::zero()` instead of `None`.

Reply:
`default` is fine w me